kernel/rtlil.cc

```diff
@@ -1565,13 +1565,21 @@ void RTLIL::Module::remove(const pool<RTLIL::Wire*> &wires)
 
 void RTLIL::Module::remove(RTLIL::Cell *cell)
 {
+	auto it = cells_.find(cell->name);
+	log_assert(it != cells_.end());
+	remove(it);
+}
+
+dict<RTLIL::IdString, RTLIL::Cell*>::iterator RTLIL::Module::remove(dict<RTLIL::IdString, RTLIL::Cell*>::iterator it)
```

Comment:
@cliffordwolf says `dict` and `pool` containers allow erasure during iteration meaning we don't need this...